src/fileioc/fileioc.asm

```diff
@@ -1573,6 +1626,29 @@ util_set_offset:
 	ld	(hl), bc
 	ret
 
+util_Arc_Unarc: ;properly handle garbage collects :P
+	call	_ChkFindSym
+	push	hl
+	call	_ChkInRAM
+	pop	hl
+	jr	nz,.arc_unarc ;if the file is already in archive, we won't trigger a gc
+	call	_LoadDEInd_s
+	ld	hl,12
+	add	hl,de
+	call	_FindFreeArcSpot ;check if we will trigger a gc
+	jr	nz,.arc_unarc ;gc will not be triggered
+	call	util_pre_gc_default_handler
+util_pre_gc_handler:=$-3
+	or	a,a
+	ret	nz ;exit if the handler returns a non-zero value
```

Comment:
One last comment, I swear.

Do people think it would make more sense for this condition to be reversed? Previously, the return type was an integer, in which case it made sense for non-zero to signal error/abort. But now that the return type is a `bool`, I feel like it makes more sense for `false` (zero) to signal error/abort.

Reply:
I agree that should probably be a ret z. The docs would have to be updated as well.

Reply:
Fixed and updated.